EIPS/eip-1062.md

```diff
@@ -0,0 +1,87 @@
+
+## Motivation
+To build fully decentralized web service, it’s necessary to have a decentralized file sotrage system. Here comes the IPFS, for three following advantages :
+- Address large amounts of data, and has unique cryptographic hash for every records.
+- Since IPFS is also based on peer to peer network, it can be really helpful to delivers large amounts of data to users, with safer way and lower the millions of cost for the bandwidth.
+- IPFS stores files in high efficient way via tracking version history for every file, and removing the duplications across the network.
+  
+Those features makes perfect match for integrating into ENS, and these make users can easily access content through ENS, and show up in the normal browser.
+
+
+## Specification
+The condition now is that the IPFS file fingerprint using base58 and in the meantime, the Ethereum using hex encryption. These comes the two restrictions :
+- Different data type, one is string, the other is integer.
+- The way to process the condition requires not only we need to transfer from IPFS to Ethereum, but also need to convert it back.
+  
+To solve these requirements, we can use binary buffer briding that gap.  
+When mapping the IPFS base58 string to ENS resolver, first we convert the Base58 to binary buffer, turn the buffer to hex encrypted format, and save to the contract. Once we want to get the IPFS resources address represented by the specific ENS, we can first find the mapping information stored as hex format before, extract the hex format to binary buffer, and finally turn that to IPFS Base58 address string.
+
+
+## Rationale
+To implement the specification, need two methods from ENS public resolver contract, when we want to store IPFS file fingerprint to contract, convert the Base58 string identifier to the hex format and invoke the `setContent` method below :
+  
+```
+function setContent(bytes32 node, bytes32 hash) public only_owner(node);
+```
+  
+Whenever users need to visit the ENS content, we call the `content` method to get the IPFS hex data, transfer to the Base58 format, and return the IPFS resources to use.
+  
+```
+function content(bytes32 node) public view returns (bytes32);
+```
+
+## Test Cases
+
+To implement the way to transfer from base58 to hex format and the reverse one, using the ‘multihashes’ library to deal with the problem.  
+The library link : [https://www.npmjs.com/package/multihashes](https://www.npmjs.com/package/multihashes)  
+To implement the method transfer from IPFS(Base58) to hex encryption format :
+  
+```
+import multihash from 'multihashes'
+
+export const toHex = function(ipfsHash) {
+ let buf = multihash.fromB58String(ipfsHash)
+ let digest = multihash.decode(buf).digest
```

Comment:
Doing this hardcodes the use of SHA2-256. Could we just store the binary multihash in a `bytes` field instead, which would provide forward compatibility?

This would be useful for swarm, too, who want to be able to store hash+key pairs in ENS.

Reply:
That looks like it would work, yes. It looks like `multihashes` has `toHexString` and `fromHexString` too, though, which would probably be a better fit.